elixir/apps/domain/test/domain/resources_test.exs

```diff
@@ -1479,71 +1479,35 @@ defmodule Domain.ResourcesTest do
 
       attrs = %{"address" => "foo"}
 
-      assert {:replaced, updated_resource, created_resource} =
-               update_or_replace_resource(resource, attrs, subject)
+      assert {:updated, updated_resource} =
+               update_resource(resource, attrs, subject)
 
       flow_id = flow.id
       updated_resource_id = updated_resource.id
       assert_receive {:expire_flow, ^flow_id, _client_id, ^updated_resource_id}
       assert_receive {:delete_resource, ^updated_resource_id}
-
-      created_resource_id = created_resource.id
-      assert_receive {:create_resource, ^created_resource_id}
-    end
-
-    test "replaces resource policies when resource is replaced", %{
-      account: account,
-      resource: resource,
-      subject: subject
-    } do
-      policy1 = Fixtures.Policies.create_policy(account: account, resource: resource)
-      policy2 = Fixtures.Policies.create_policy(account: account, resource: resource)
-
-      :ok = Domain.Policies.subscribe_to_events_for_account(account)
-
-      attrs = %{"address" => "foo"}
-
-      assert {:replaced, _updated_resource, created_resource} =
-               update_or_replace_resource(resource, attrs, subject)
-
-      assert Repo.get_by(Domain.Policies.Policy, id: policy1.id).deleted_at
-      assert Repo.get_by(Domain.Policies.Policy, id: policy2.id).deleted_at
-
-      assert_receive {:delete_policy, deleted_policy_id}
-      assert deleted_policy_id in [policy1.id, policy2.id]
-
-      assert_receive {:delete_policy, deleted_policy_id}
-      assert deleted_policy_id in [policy1.id, policy2.id]
-
-      assert_receive {:create_policy, created_policy_id}
-
-      assert Repo.get_by(Domain.Policies.Policy, id: created_policy_id).resource_id ==
-               created_resource.id
-
-      assert_receive {:create_policy, created_policy_id}
-
-      assert Repo.get_by(Domain.Policies.Policy, id: created_policy_id).resource_id ==
-               created_resource.id
+      assert_receive {:create_resource, ^updated_resource_id}
     end
 
-    test "replaces the resource when type is changed", %{resource: resource, subject: subject} do
+    test "updates the resource when type is changed", %{resource: resource, subject: subject} do
       attrs = %{"type" => "ip", "address" => "10.0.10.1"}
 
-      assert {:replaced, updated_resource, created_resource} =
-               update_or_replace_resource(resource, attrs, subject)
+      assert {:updated, updated_resource} =
+               update_resource(resource, attrs, subject)
 
```

Comment:
```suggestion
      assert updated_resource.id == resource.id
```

Reply:
👍 This suggestion was taken, just moved one line down